[13.0][MIG] l10n_ch_base_bank

i-vyshnevska opened the pull request:
migrate l10n_ch_base_bank
~it blocked by https://github.com/odoo/odoo/pull/37651~
it blocked by  https://github.com/odoo/odoo/pull/49431

nbessi committed:
[IMP] refactoring module structure

nbessi committed:
[FIX] port of base bank

nbessi committed:
[REF] bvr report

nbessi committed:
[REF] bvr duplicated stuff

nbessi committed:
[REF] move from spli to bank"

nbessi committed:
[FIX] non serializable type

nbessi committed:
[FIX] default in sale workflow fo generated invoices

nbessi committed:
[TYPO]

nbessi committed:
[REFACTOR] bank dispertion

nbessi committed:
[FIX] contributors list

nbessi committed:
[FIX] encoding metadata

nbessi committed:
[ADD] add post related field

nbessi committed:
[IMP] add notion of CCP on bank instead of having an account on res.partner.bank.
The code is retro compatible and if you do not have a ccp on bank it will act as before

nbessi committed:
[IMP] change version of addons + update changelog

nbessi committed:
[FIX] res.partner.bank view

nbessi committed:
[FIX] ccp related is read only

nbessi committed:
[FIX] blank line

nbessi committed:
[FIX] bank view city and zip where duplicated

nbessi committed:
[FIX] hide ccp if the bic is the postfinance one

nbessi committed:
[FIX] bank search and name get. Add a intelligent name search and add city to name get

nbessi committed:
[ADD] street on res.bank name_get and name_search

vrenaville committed:
[FIX]only bvr requiered bvr ref

jgrandguillaume committed:
[IMP] Add all translations .pot file in order to allow people to translate module on LP

Commit:
fix a bug when an account number is empty

Commit:
Launchpad automatic translations update.

nbessi committed:
Set all addons to uninstallable

nbessi committed:
Rename all addons to xxx_unported

PCatinean committed:
[ADD] Added l10n_ch_base_bank migrated to the new api

nbessi committed:
Cleanup of l10n_ch_base_bank port

nbessi committed:
Remove deprecated size

nbessi committed:
Better variable name

nbessi committed:
Renaming variables

nbessi committed:
Better function call

nbessi committed:
Remove faulty use of _ids

nbessi committed:
Typo

nbessi committed:
Fix wrong error messages

nbessi committed:
Fix forgotten return statement

nbessi committed:
Fix exceptions

nbessi committed:
Fix exceptions

nbessi committed:
Improve error message anf description string

nbessi committed:
Fix invalid variable name

nbessi committed:
Fix base bank constrains and add unit test

nbessi committed:
Add fast suite

nbessi committed:
Fix on change partner_id

Commit:
Constrains must includes only fields of the current model

Anyway, the check already exist on the res.bank model.  And we want to check
the ccp duplication when it is changed or the bank is changed, so
('acc_number', 'bank') is correct.

nbessi committed:
Removing deprecated fast_suite

nbessi committed:
Update addons version

Commit:
Update translations in l10n_ch_base_bank

Commit:
Remove view that breaks the onchange because it replaces the whole field

Remove it entirely because 'select="1"' has no effect on views now that we use
search views.

Commit:
Do not use 'replace' position which can lose attributes

Only change the domain attribute

Commit:
Remove deprecated type of view

Commit:
onchange_partner_bank breaks the base onchange

Because it is declared as api.onchange whereas the base method is an
api.multi with the old-api style 'on_change' in XML. If we declare
onchange_partner_bank as api.onchange, it overrides the base one and
that's not what we want.

nbessi committed:
Fix faulty call to super

yvaucher committed:
Add license key in __openerp__.py

nbessi committed:
Replace PIL rendered with Reportlab rendered

It allows to have lighter, faster, vectorial payment slip.

This should fix most of current performance and OCR issues

gurneyalex committed:
Add OCA as author of OCA addons

In order to get visibility on https://www.odoo.com/apps the OCA board has
decided to add the OCA as author of all the addons maintained as part of the
association.

vrenaville committed:
[FIx] check the bvr reference for bvr supplier invoice

oca-transbot committed:
OCA Transbot updated translations from Transifex

sbidoul committed:
[UPD] prefix versions with 8.0

pedrobaeza committed:
[MIG] Make modules uninstallable

yvaucher committed:
[PORT] l10n_ch_base_bank to 9.0
- Reactivate model
- move description in README.rst
- bump version
- move files in folders models, views, data
- add tests
- adapt the logic to new API 9.0, now bank account type are computed.
Thus we define if a bank account is of type Swiss postal account.
Plus on invoice, detect BVR reference and do check bank account is
configured to handle BVR reference (is a CCP or Bank/CCP)
- Update license headers where year was missing

yvaucher committed:
Update translations

yvaucher committed:
Explain why bvr_adherent_num is not present in view

yvaucher committed:
[PORT] l10n_ch_base_bank - accept IBAN for postal account + auto detect postal account type to set bank

yvaucher committed:
[PORT] l10n_ch_base_bank - remove constraint on ccp, a ccp can be use by
the same bank on different adresses
CCP is now computed
Plus IBAN postal account are transformed to CCP into ccp field

BT-ojossen committed:
return list of all banks if name is empty

otherwise the customer thinks that there aren't any banks in the system.

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

alexis-via committed:
Re-write l10n_ch_sepa for Odoo 9, building above account_banking_sepa_credit_transfer

alexis-via committed:
Should now generate VALID pain.001.001.03.ch.02 files

alexis-via committed:
Remove useless inherit of invoice form view (which broke the inherit in the module account_payment_partner)

alexis-via committed:
Remove a method that has nothing to do in a localization module, and conflicts with account_payment_partner

alexis-via committed:
Fix BIC length

alexis-via committed:
Fix unit-test suite of l10n_ch_base_bank

oca-transbot committed:
OCA Transbot updated translations from Transifex

yvaucher committed:
[TYPO]

oca-transbot committed:
OCA Transbot updated translations from Transifex

pedrobaeza committed:
[MIG] Rename manifest files

yvaucher committed:
Fix ccp computation from IBAN of postal account by normalizing it
This allows to put IBAN with spaces separations for readability and visual check

yvaucher committed:
Improve error message and make it translatable

grindtildeath committed:
Changed version in Manifest to 10.0.1.0.1

grindtildeath committed:
Fix import error on base_iban module

grindtildeath committed:
Change import openerp to odoo

grindtildeath committed:
Changes in README and fix of version

yvaucher committed:
Improve UX by defining onchange to set bank when account is entered
Set the bank when possible
and set ccp when undefined
Bank is defined as:
- Found bank with CCP matching Bank CCP
- Swiss post when CCP is no matching a Bank CCP
- Found bank by clearing when using iban
For CCP it can be:
- a postal account, we copy acc_number
- a postal account in iban format, we transform acc_number
- a bank account with CCP on the bank, we use ccp of the bank
- otherwise there is no CCP to use

Commit:
Allow whitespaces in search of invoice references

When searching in Vendor references, we want to ignore the whitespaces.
This is because very often, it will contain BVR/ESR number that contains
whitespaces. And the user usually searches for the invoice number which
is only a part of the BVR/ESR number.

Example:
* the vendor reference on the supplier invoice is: 27 29990 00000 00001 70400 25019
* the user knows the invoice's supplier reference: 17040025
* when searching 17040025, he expects to find the invoice
* the reverse situation, where the ref contains no whitespaces and the
  user searches for a string with whitespaces might also happen

Commit:
Remove support of strict operators

Only the like (&co.) operators will do the fuzzy search without
whitespaces. Strict equality operators will remain strict, which means
they will need to have an exact match. This what we would the most
expect from '='. Users will use the like/ilike from the UI and that's
only where whe should be tolerant regarding extra or missing whitespaces.

It fixes along the way:

* =like and =ilike where not working as they are not SQL operators
* when the original value only contains spaces, it would generate a
  costly query that matches all the records, now there is a fast path

yvaucher committed:
Adapt check on invoice BVR ref

yvaucher committed:
Fix onchange bank on partner bank when acc_number is undefined (#318)

Martronic-SA committed:
Contraint on BVR adherent number / CPP pair

First of all there is a confusion for the BVR coding line:
0100000600007>103559000000000020170013270+ 100007254>
              ^ reference beginning with   ^ adherent number
                bank client account number
                ( 6 chars )
You are naming the 6 chars client account number as BVR adherent number which is false.
Here Raiffeisen documentation : https://www.raiffeisen.ch/content/dam/www/rch/pdf/firmenkunden/zahlungsverkehr-und-konten/fr/fiche-produit-bulletin-versement-orange.pdf
However, both of these numbers can be used many times but the pair (adherent number / client account number) must be unique.

yvaucher committed:
[MIG] l10n_ch_base_bank: Migration to 11.0

Remove coding UTF-8 from headers
Reset version to 11.0.1.0.0
Update README.rst
Remove deprecated api.one

yvaucher committed:
Bypass wrong positive sql-injection pylint error as operator is already safely checked

yvaucher committed:
Rename BVR and BVR/ESR to ISR in english and make it the base term

yvaucher committed:
Rename BVR and BVR/ESR to ISR in english and make it the base term

yvaucher committed:
Fix search invoice with =operators

yvaucher committed:
Fix tests when no partner is set on bank account (due to default value)

oca-transbot committed:
OCA Transbot updated translations from Transifex

yvaucher committed:
Add migration script

oca-transbot committed:
OCA Transbot updated translations from Transifex

asaunier committed:
[IMP] l10n_ch_base_bank: bic and city required on view for switzerland (#407)

* [IMP] l10n_ch_bank: bic and city required on view for switzerland

updated csv filling the country column

asaunier committed:
[11.0] Fix incorrect XML in l10n_ch_base_bank view (#414)

* Rename openerp element to odoo

* Fix incorrect XML in l10n_ch_base_bank view

Commit:
[11.0][FIX] _compute_acc_type failing on recordsets

sbidoul committed:
remove obsolete .pot files [ci skip]

oca-travis committed:
[UPD] Update l10n_ch_base_bank.pot

OCA-git-bot committed:
[MIG] Make modules uninstallable

chrispi-ch committed:
Translated using Weblate (German)

Currently translated at 100.0% (25 of 25 strings)

Translation: l10n-switzerland-11.0/l10n-switzerland-11.0-l10n_ch_base_bank
Translate-URL: https://translation.odoo-community.org/projects/l10n-switzerland-11-0/l10n-switzerland-11-0-l10n_ch_base_bank/de/

MaxyMoos committed:
[FIX] res.partner.bank: retrieve partner_name from context on creation (#428)

asaunier committed:
[MIG] l10n_ch_base_bank: Migration to odoo 12

Commit:
l10n_ch_base_bank: update tests and invoice model

Changet in test new() to create and write() to run test properly
update tetst cases due to errors, refactor behavior of updating
acc_number and ccp

Due to updates in  account_payment_partner 95174ee
changet from direct access to method partner_banks_to_show

Commit:
[IMP] use common method from bank-payment for bank searching

oca-travis committed:
[UPD] Update l10n_ch_base_bank.pot

Commit:
[IMP] cleanup pylint

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update l10n_ch_base_bank.pot

Commit:
[FIX] drop never running logig, cleanup

Commit:
[FIX] update l10n_ch_base_bank to respect SEPA logic

OCA-git-bot committed:
[UPD] README.rst

svalaeys committed:
Translated using Weblate (French)

Currently translated at 100.0% (25 of 25 strings)

Translation: l10n-switzerland-12.0/l10n-switzerland-12.0-l10n_ch_base_bank
Translate-URL: https://translation.odoo-community.org/projects/l10n-switzerland-12-0/l10n-switzerland-12-0-l10n_ch_base_bank/fr/

yvaucher committed:
Remove support for 6 digits postal account number

yvaucher committed:
Fix typo on error message

oca-travis committed:
[UPD] Update l10n_ch_base_bank.pot

yvaucher committed:
Use Odoo core field l10n_ch_postal to replace ccp

Plus stop trying to get bank from postal number thus remove
the ccp fields on res.bank

This will prepare the migration to version 13.0

yvaucher committed:
Remove `get_number`

OCA-git-bot committed:
[UPD] README.rst

Commit:
[FIX] Resolve conflicts

Commit:
[MIG][13.0] l10n_ch_base_bank migrate

yvaucher commented:
@i-vyshnevska brand new pre-commit config is there on the project! please rebase

> pre-commit has been adopted by the OCA and installed to this repository.
> All 13.0 branch of the OCA now comes with pre-commit hook.
> 
> You might have to run pre-commit run -a to auto-fix what needs to be fixed.
> 
> For further development on branch 13.0 on any OCA repository pre-commit hooks will be available and can be installed with pip install pre-commit then pre-commit install.

Commit:
[IMP] l10n_ch_base_bank: apply black

TDu committed:
Fix clearing number extraction

The clearing number in IBAN is padded with zero on the left.
They need to be removed when extracting it as to have the correct one.

yvaucher commented:
@i-vyshnevska please cherry-pick the following commit:

https://github.com/OCA/l10n-switzerland/pull/479/commits/b18f94a1518404a0fbebac78400f232e03325dc4

Commit:
[IMP] aplly pre-commit

yvaucher committed:
Remove duplicate l10n_ch_postal field on bank account

i-vyshnevska committed:
Merge pull request #1 from yvaucher/13-l10n_ch_base_bank

Remove duplicate l10n_ch_postal field on bank account

yvaucher commented:
superseeded by #550
